agent: add support to FQDN in ManagerHost

dougsland opened the pull request:
GitHub-Issue: https://github.com/containers/hirte/issues/276

dougsland committed:
agent: add support to FQDN in ManagerHost

GitHub-Issue: https://github.com/containers/hirte/issues/276

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
network: move typesafe_inet_ntop4 to network module

As an effort to move the network bits to the same place.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
testing: add tests for functions in network

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

engelmi commented on src/libhirte/test/common/network/get_address_test.c:
```diff
@@ -0,0 +1,48 @@
+
+#include "libhirte/common/network.h"
+
+#define bool_to_str(b) b ? "true" : "false"
```
Nit: `bool_to_str` was only used in `string-utils_test` before. We could move it now to a central location, maybe [libhirte/common/string-utils](https://github.com/containers/hirte/blob/main/src/libhirte/common/string-util.h) or [libhirte/common/common)(https://github.com/containers/hirte/blob/main/src/libhirte/common/common.h)?

engelmi commented on src/libhirte/common/network.c:
```diff
@@ -0,0 +1,67 @@
+        if (res->ai_family == AF_INET) { // IPv4 address
+                struct sockaddr_in *ipv4 = (struct sockaddr_in *) res->ai_addr;
+                addr = &(ipv4->sin_addr);
+                if (inet_ntop(AF_INET, addr, ip_address, ip_address_size) == NULL) {
```
Nit: Could be replaced with the `typesafe_inet_ntop4`.

engelmi commented on src/libhirte/common/network.c:
```diff
@@ -0,0 +1,67 @@
+
+        int status = getaddrinfo(domain, NULL, &hints, &res);
+        if (status != 0) {
+                fprintf(stderr, "getaddrinfo failed: %s\n", gai_strerror(status));
```
Nit: This and the following `fprint`s can be replaced by `hirte_log_errorf`.

engelmi reviewed (commented):
Looks great! I only have a few nit-picky comments. 

And thanks for adding unit tests! Having one test file for one function under test seems better than one test file containing multiple (as it is currently - we'll have to refactor that).

dougsland committed:
string-util: add bool_to_str

More modules are using bool_to_str, adding to a module.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
network: use typesafe_inet_ntop family

convert the code to use typesafe_inet_ntop function.
Also created typesafe_inet_ntop6 regarding IPv6.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
network: move assemble_tcp_address to network module

make sure we have network bits in related source/tests.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
network: add _cleanup_free_ macro

It was missing a _cleanup_free_ macro, good catch valgrind!

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

engelmi reviewed (approved):
LGTM
